crates/sui-transactional-test-runner/src/programmable_transaction_test_parser/token.rs

```diff
@@ -0,0 +1,165 @@
+    Void,
+}
+
+pub const TRANSFER_OBJECTS: &str = "TransferObjects";
+pub const SPLIT_COIN: &str = "SplitCoin";
+pub const MERGE_COINS: &str = "MergeCoins";
+pub const MAKE_MOVE_VEC: &str = "MakeMoveVec";
+pub const GAS_COIN: &str = "Gas";
+pub const INPUT: &str = "Input";
+pub const RESULT: &str = "Result";
+pub const NESTED_RESULT: &str = "NestedResult";
+
+impl Display for CommandToken {
+    fn fmt<'f>(&self, formatter: &mut fmt::Formatter<'f>) -> Result<(), fmt::Error> {
+        let s = match *self {
+            CommandToken::Whitespace => "[whitespace]",
+            CommandToken::Comment => "[comment]",
+            CommandToken::Ident => "[identifier]",
+            CommandToken::Number => "[num]",
+            CommandToken::ColonColon => "::",
+            CommandToken::Colon => ":",
+            CommandToken::Comma => ",",
+            CommandToken::Semi => ";",
+            CommandToken::LBracket => "[",
+            CommandToken::RBracket => "]",
+            CommandToken::LParen => "(",
+            CommandToken::RParen => ")",
+            CommandToken::TypeArgString => "<...>",
+            CommandToken::Void => "[void]",
+        };
+        fmt::Display::fmt(s, formatter)
+    }
+}
+
+impl Token for CommandToken {
+    fn is_whitespace(&self) -> bool {
+        matches!(self, Self::Whitespace | Self::Comment | Self::Void)
+    }
+
+    fn next_token(s: &str) -> anyhow::Result<Option<(Self, usize)>> {
+        // parses a string where start matches end.
+        // performs simple matching for start/end pairs
+        fn parse_sub_token_string(
+            s: &str,
+            start: char,
+            end: char,
+        ) -> anyhow::Result<Option<usize>> {
+            // the length of the string until the matching end
+            let mut len = 0;
+            // the count of number of active start/end pairs
+            let mut count = 0i32;
+            let mut chars = s.chars().peekable();
+            loop {
+                let Some(c) = chars.next() else {
+                    bail!("Unexpected end of string after '('")
+                };
+                len += 1;
+                if c == start {
+                    // a new start char
+                    count += 1
+                } else if c == end {
+                    // an end char
+                    count -= 1;
+                    if count == 0 {
+                        // end found
+                        break;
+                    }
+                }
+            }
+            Ok(Some(len))
+        }
+
+        // type arguments get delegated to a different parser
+        if s.starts_with('<') {
+            return Ok(parse_sub_token_string(s, '<', '>')?.map(|len| (Self::TypeArgString, len)));
+        }
+        // comments
+        if s.starts_with("//") {
+            let end = s.find('\n').map(|end| end + 1).unwrap_or(s.len());
+            return Ok(Some((Self::Comment, end)));
+        }
+        if s.starts_with("/*") {
```

Comment:
Not super important, but maybe worth commenting here that this doesn't support nested `/*` comments.